src/embedding/ad.ts

```diff
@@ -0,0 +1,417 @@
+// (C) 2020 GoodData Corporation
+import get = require('lodash/get');
+import {
+    CommandFailed,
+    IObjectMeta,
+    IGdcMessageEvent,
+    IGdcMessageEventDataWrapper,
+    CommandFailedData,
+    isCommandFailedData,
+    GdcProductName
+} from './common';
+import { VisualizationObject } from '../VisualizationObject';
+
+export interface IBaseExportConfig {
+    title?: string;
+    format?: 'xlsx' | 'csv' | 'raw';
+    mergeHeaders?: boolean;
+}
+
+// NOTE: AFM is ommitted on purpose; it should be added by AD itself; create new type using Omit<>
+export interface IExportConfig extends IBaseExportConfig {
+    showFilters?: boolean;
+}
+
+export namespace EmbeddedAnalyticalDesigner {
+    export type IADMessageEvent<T, TBody> = IGdcMessageEvent<GdcProductName.ANALYTICAL_DESIGNER, T, TBody>;
+    export type IAdGdcMessageEventDataWrapper<T, TBody> = IGdcMessageEventDataWrapper<
+        GdcProductName.ANALYTICAL_DESIGNER,
+        T,
+        TBody
+    >;
+
+    export enum GdcAdCommandType {
+        DrillableItems = 'drillableItems',
+        OpenInsight = 'openInsight',
+        Save = 'saveInsight',
+        SaveAs = 'saveAsInsight',
+        Export = 'exportInsight',
+        Undo = 'undo',
+        Redo = 'redo'
+    }
+
+    export enum GdcAdEventType {
+        ListeningForDrillableItems = 'listeningForDrillableItems',
+        NewInsightInitialized = 'newInsightInitialized',
+        InsightOpened = 'insightOpened',
+        // use `visualizationSaved` because of backward compatibility
+        InsightSaved = 'visualizationSaved',
```

Comment:
We should attach to this [guide](https://help.gooddata.com/doc/en/building-on-gooddata-platform/gooddata-integration-into-your-application/embed-gooddata-elements-into-your-applications/capture-the-save-an-insight-event-from-embedded-analytical-designer#CapturetheSaveanInsightEventfromEmbeddedAnalyticalDesigner-Example:EmbeddedAnalyticalDesignerwithEventCatching)

Reply:
updated